src/components/navigation/link.ts

```diff
@@ -231,31 +243,75 @@ export class Link {
   }
 
   private static determineVisibility({
+    hasEntitlement,
     visibilitySettings,
     brand,
     userType,
+    userEntitlements,
+    entitlementConfig,
   }: {
+    hasEntitlement: boolean;
     visibilitySettings: VisibilitySettings;
     brand: Brand;
     userType?: UserTypeExternal.Guest | MappedUserType;
+    userEntitlements: string[];
+    entitlementConfig?: EntitlementConfig;
   }) {
     if (!visibilitySettings.brand[brand]) {
       return false;
     }
 
-    if (
-      userType === UserTypeExternal.Guest &&
-      visibilitySettings.userType &&
-      visibilitySettings.userType[userType] !== undefined
-    ) {
+    const hasRestrictedEntitlement = Link.hasRestrictedEntitlement({
+      entitlementConfig,
+      userEntitlements,
+    });
+
+    if (hasRestrictedEntitlement) {
+      return false;
+    }
+
+    const hideIfRequiredEntitlementIsMissing =
+      Link.hideIfRequiredEntitlementIsMissing({
+        hasEntitlement,
+        entitlementConfig,
+      });
+
+    if (hideIfRequiredEntitlementIsMissing) {
+      return false;
+    }
+
+    // This scenario is essential when dealing with guest users.
+    // The guest user type was introduced at a later stage.
+    // Rather than modifying visibility settings for the guest user type across all link nodes,
+    // our goal is to selectively hide a single link in the navigation for guest users.
```

Comment:
FYI, `stkb.rewrap` is a great vsc extension for line wrapping which makes handling mutli-line comments a breeze.

Reply:
cool, didn't know about it. Thanks for sharing it.